### PR TITLE
fix(cli): auth file should be chmod 600

### LINKED ISSUE #6911: The auth.yml file generated by Immich CLI is world readable
### The bug

Using the `immich login-key` command for Immich CLI generates the file `~/.config/immich/auth.yml`. This file contains the sensitive API key, but the file is world readable by default.

```sh
ls -l ~/.config/immich/
total 4
-rw-r--r-- 1 jordan jordan 97 Feb  4 09:37 auth.yml
```

### The OS that Immich Server is running on

N/A

### Version of Immich Server

N/A

### Version of Immich Mobile App

N/A

### Platform with the issue

- [ ] Server
- [ ] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
N/A
```

### Your .env content

```Shell
N/A
```

### Reproduction steps

```bash
1. Use `immich login-key` to authenticate to the Immich instance.
```

### Additional information

Immich CLI is version 2.0.6.

### PR DESCRIPTION
**Pull request by etnoy**
Fixes #6911 

* Creates auth.yml with chmod 600
* E2e tests for the correct chmod

### CODE REVIEW

File: cli/test/e2e/login-key.e2e-spec.ts [hunk 1]
```diff
@@ -28,6 +32,12 @@ describe(`login-key (e2e)`, () => {
 
     const api = await setup();
     apiKey = api.apiKey;
+
+    await unlink(TEST_AUTH_FILE).catch(() => {});
+  });
+
+  afterEach(async () => {
+    await unlink(TEST_AUTH_FILE).catch(() => {});
```
Comment (danieldietzler) [inline cli/test/e2e/login-key.e2e-spec.ts:40]: Probably just me being confused but why do we need to unlink the file at the beginning and the end of each test? Isn't beginning enough?

**Review by danieldietzler — APPROVED** [review]

File: cli/test/e2e/login-key.e2e-spec.ts (hunk 1 above)
Comment (etnoy) [reply cli/test/e2e/login-key.e2e-spec.ts:40]: This is just to tidy up for the next test suite. Maybe I can move it to the afterAll instead

File: cli/test/e2e/login-key.e2e-spec.ts [hunk 2]
```diff
@@ -39,4 +49,27 @@ describe(`login-key (e2e)`, () => {
...
+
+    expect(parsedConfig).toEqual(expect.objectContaining({ instanceUrl, apiKey }));
+  });
+
+  it('should create an auth file with chmod 600', async () => {
+    await new LoginCommand(CLI_BASE_OPTIONS).run(instanceUrl, apiKey);
+
+    const data = await stat(TEST_AUTH_FILE);
+
+    expect(convert.object(data.mode)).toEqual(
```
Comment (benmccann) [inline cli/test/e2e/login-key.e2e-spec.ts:67]: maybe we could check the permissions without adding a new dependency? I just did a quick google and it looks pretty straightforward: https://www.martin-brennan.com/nodejs-file-permissions-fstat/

File: cli/test/e2e/login-key.e2e-spec.ts (hunk 1 above)
Comment (etnoy) [reply cli/test/e2e/login-key.e2e-spec.ts]: done

File: cli/test/e2e/login-key.e2e-spec.ts (hunk 2 above)
Comment (etnoy) [reply cli/test/e2e/login-key.e2e-spec.ts]: done, thanks

**Pull request merged**